subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -2414,26 +2414,58 @@ static uint8_t tx_cmplt_get(uint16_t *handle, uint8_t *first, uint8_t last)
 			struct pdu_data *p;
 			uint8_t fragments;
 
+			/* NOTE: tx_cmplt_get() is permitted to be called
+			 *       multiple times before the tx_ack queue which is
+			 *       associated with Rx queue is changed by the
+			 *       dequeue of Rx node.
+			 *
+			 *       Tx node is released early without waiting for
+			 *       any dependency on Rx queue. Released Tx node
+			 *       reference is overloaded to store the Tx
+			 *       fragment count.
+			 *
+			 *       A hack is used here that depends on the fact
+			 *       that memory addresses have a value greater than
+			 *       0xFF, to determined if a node Tx has been
+			 *       released in a prior iteration of this function.
```

Comment:
So is this a temporary solution? If so, I suggest to create a GH issue to track this and to implement a proper solution.

Reply:
Is implementing this without a "hack" a large task, or what is the reasoning to use a "hack"?

Reply:
The implementation comes from the original controller contribution, and has not been changed since then.

Reply:
I will give a thought, this has to done for the ACL Tx ack too where there is difference between ack that is generating a num complete only for data PDU and not for control PDU (inline ENC_REQ PDU) that was transmitted.

Reply:
I have an alternative solution for which I will make a PR today

Reply:
@mtpr-ot should we merge this PR and thereafter yours? (#49855 LGTM, though you apply the same changes to ACL too).

Reply:
@cvinayak ACL, what do you mean?

Reply:
I think my solution is more intuitive, and if possible, should go in first. You need to re-implement this for current state of main anyway.

Reply:
> ACL, what do you mean? 

Check this PR changes, ACL connections too use similar bitmask for num complete counting, it will be best if both are fixed.

> I agree, this is very difficult to understand and maintain, so instead of repeating the pattern from node_tx for node_tx_iso, could we perhaps use the available unused bits in node_tx_iso for a state, which can maintain the information instead of overloading the pointer meaning?

#49855 is still the same hack but hiding the code with macros, compared to your original request in this PR was for a redesign.

There seems to be no merge conflicts reported in this PR. As #49855 is just a refactor of the existing hack, it should go after this PR which is a fix for a github issue.

Reply:
@cvinayak Please rebase